### DISCUSSION TITLE
client exited even though serviceReady

### DISCUSSION THREAD

**Discussion by marekzan** [mfussenegger/nvim-jdtls#309] [Q&A] [answered]
When I open a .java file I get an immediate Error

`Client X quit with exit code 1 and signal 0`

but after that I see "Starting Java language server" with "ServiceReady" in the end.

The logs output via JdtShowLogs is the following:

```
[ERROR][2022-08-20 15:01:47] .../vim/lsp/rpc.lua:439	"rpc"	"java"	"stderr"	"Error: Unable to access jarfile $JDTLS_HOME/plugins/org.eclipse.equinox.launcher_*.jar\n"
[ERROR][2022-08-20 15:01:47] .../vim/lsp/rpc.lua:439	"rpc"	"java"	"stderr"	"WARNING: Using incubator modules: jdk.incubator.foreign, jdk.incubator.vector\n"
```

And I have to say that I am only able to use `JdtShowLogs` and `JdtWipeDataAndRestart` Commands.
`JdtSetRuntime` for example yields `Not an editor command:`

`:LspInfo` is showing:

![image](https://user-images.githubusercontent.com/5881806/185747889-64863464-e0bd-48c1-a8d4-81dbc51c27a9.png)

Is it normal that `filetypes` is empty on the `jdt.ls` client? 
and is it normal that it is saying `jdt.ls` instead of `jdtls` as it is showing in the `other clients that match the filetype`?

I installed jdtls via Mason and my config is in the ftplugin folder as java.lua file:

![image](https://user-images.githubusercontent.com/5881806/185748071-cde7b8af-7ee2-49fe-8d82-651b8acc58ab.png)

I adjusted the `-jar`, the `-configuration` and the `-data` accordingly 

None of the `jdtls` options are working like `organize_imports()`.
I am not getting any IntelliSense or `lsp.buf.hover()` functionality.

Any help would be very appreciated!

**Comment by mfussenegger** [comment, marked as answer]
Sounds like you're using both lspconfig.jdtls.setup and jdtls.  That doesn't work, see https://github.com/mfussenegger/nvim-jdtls#nvim-lspconfig-and-nvim-jdtls-differences

> Is it normal that filetypes is empty on the jdt.ls client?
> and is it normal that it is saying jdt.ls instead of jdtls as it is showing in the other clients that match the filetype?

Yes, see the link above. nvim-jdtls doesn't setup filetype hooks on its own, that's what your `ftplugin/java.lua` file is for.

**Reply by marekzan** [reply]
Thank you very much for your help. This solved my issue!
